src/enftun_xtt.c

```diff
@@ -0,0 +1,894 @@
+    if(write_ret < 0) {
+        return SAVE_TO_FILE_ERROR;
+    }
+    if (0 != xtt_crypto_memcmp(xtt_null_identity.data, requested_client_id->data, sizeof(xtt_identity_type))) {
+        printf("(I requested id: {");
+        for (size_t i=0; i < sizeof(xtt_identity_type); ++i) {
+            printf("%#02X", requested_client_id->data[i]);
+            if (i < (sizeof(xtt_identity_type)-1)) {
+                printf(", ");
+            } else {
+                printf("}\n");
+            }
+        }
+    }
+
+    // 2) Get longterm keypair
+    xtt_ecdsap256_pub_key my_longterm_key = {.data = {0}};
+    if (XTT_RETURN_SUCCESS != xtt_get_my_longterm_key_ecdsap256(&my_longterm_key, ctx)) {
+        printf("Error getting my longterm key!\n");
+        return 1;
+    }
+    printf("My longterm key: {");
+    for (size_t i=0; i < sizeof(xtt_ecdsap256_pub_key); ++i) {
+        printf("%#02X", my_longterm_key.data[i]);
+        if (i < (sizeof(xtt_ecdsap256_pub_key)-1)) {
+            printf(", ");
+        } else {
+            printf("}\n");
+        }
+    }
+    xtt_ecdsap256_priv_key my_longterm_private_key = {.data = {0}};
+    if (XTT_RETURN_SUCCESS != xtt_get_my_longterm_private_key_ecdsap256(&my_longterm_private_key, ctx)) {
+        printf("Error getting my longterm private key!\n");
+        return 1;
+    }
+
+    // 3) Save longterm keypair as X509 certificate and ASN.1-encoded private key
+    unsigned char cert_buf[XTT_X509_CERTIFICATE_LENGTH] = {0};
+    if (0 != xtt_x509_from_ecdsap256_keypair(&my_longterm_key, &my_longterm_private_key, &my_assigned_id, cert_buf, sizeof(cert_buf))) {
+        fprintf(stderr, "Error creating X509 certificate\n");
+        return CERT_CREATION_ERROR;
+    }
+    write_ret = xtt_save_to_file(cert_buf, sizeof(cert_buf), longterm_cert_out_file);
+    if(write_ret < 0){
+        return SAVE_TO_FILE_ERROR;
+    }
+
+    unsigned char asn1_priv_buf[XTT_ASN1_PRIVATE_KEY_LENGTH] = {0};
+    if (0 != xtt_asn1_from_ecdsap256_private_key(&my_longterm_private_key, &my_longterm_key, asn1_priv_buf, sizeof(asn1_priv_buf))) {
+        fprintf(stderr, "Error creating ASN.1 private key\n");
+        return 1;
+    }
+    write_ret = xtt_save_to_file(asn1_priv_buf, sizeof(asn1_priv_buf), longterm_private_key_out_file);
+    if(write_ret < 0) {
+        return SAVE_TO_FILE_ERROR;
+    }
+
+    // 4) Get pseudonym
+    xtt_daa_pseudonym_lrsw my_pseudonym = {.data = {0}};
+    if (XTT_RETURN_SUCCESS != xtt_get_my_pseudonym_lrsw(&my_pseudonym, ctx)) {
+        printf("Error getting my pseudonym!\n");
+        return 1;
+    }
+    printf("I have pseudonym: {");
+    for (size_t i=0; i < sizeof(xtt_daa_pseudonym_lrsw); ++i) {
+        printf("%#02X", my_pseudonym.data[i]);
+        if (i < (sizeof(xtt_daa_pseudonym_lrsw)-1)) {
+            printf(", ");
+        } else {
+            printf("}\n");
+        }
+    }
+
+    return 0;
+}
+
+#ifdef USE_TPM
+static int
+read_nvram(unsigned char *out,
```

Comment:
This should definitely be in `xtt`, not here.  

Keep it here for now.  But after the next round of review on this PR, I'll want to discuss.